### PR TITLE
statistics: track used stats status during query

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/36481

Problem Summary:

### What is changed and how it works?

Track used stats during query and recorded in slow log

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -2526,6 +2526,8 @@ const (
 	SlowLogIsExplicitTxn = "IsExplicitTxn"
 	// SlowLogIsWriteCacheTable is used to indicate whether writing to the cache table need to wait for the read lock to expire.
 	SlowLogIsWriteCacheTable = "IsWriteCacheTable"
+	// SlowLogStatsLoadStatus is used to indicate the stats load status in memory
+	SlowLogStatsLoadStatus = "StatsLoadStatus"
```
Comment (time-and-fate) [inline sessionctx/variable/session.go:2530]: According to the design, we won't add a new field. We just extend the content of the current `SlowLogStatsInfoStr` field.

File: statistics/selectivity.go [hunk 2]
```diff
@@ -227,6 +227,7 @@ func (coll *HistColl) Selectivity(ctx sessionctx.Context, exprs []expression.Exp
 	for id, colInfo := range coll.Columns {
 		col := expression.ColInfo2Col(extractedCols, colInfo.Info)
 		if col != nil {
+			recordUsedItemStatsStatus(ctx, colInfo, nil, coll.PhysicalID, id, false)
```
Comment (time-and-fate) [inline statistics/selectivity.go:230]: I think we don't need to record it here.

File: statistics/selectivity.go [hunk 3]
```diff
@@ -260,6 +261,7 @@ func (coll *HistColl) Selectivity(ctx sessionctx.Context, exprs []expression.Exp
 	for id, idxInfo := range coll.Indices {
 		idxCols := FindPrefixOfIndexByCol(extractedCols, coll.Idx2ColumnIDs[id], id2Paths[idxInfo.ID])
 		if len(idxCols) > 0 {
+			recordUsedItemStatsStatus(ctx, nil, idxInfo, coll.PhysicalID, id, true)
```
Comment (time-and-fate) [inline statistics/selectivity.go:264]: ditto.

File: sessionctx/variable/session.go (hunk 1 above)
Comment (Yisaer) [reply sessionctx/variable/session.go]: updated.

File: statistics/table.go [hunk 4]
```diff
@@ -1378,3 +1386,18 @@ func CheckAnalyzeVerOnTable(tbl *Table, version *int) bool {
 	// This table has no statistics yet. We can directly return true.
 	return true
 }
+
+// recordUsedItemStatsStatus only records un-FullLoad item load status during user query
+func recordUsedItemStatsStatus(sctx sessionctx.Context, loadStatus StatsLoadedStatus,
+	tableID, id int64, isIndex bool) {
+	if sctx.GetSessionVars().InRestrictedSQL || loadStatus.IsFullLoad() {
```
Comment (time-and-fate) [inline statistics/table.go:1393]: I think we don't need to filter internal SQLs here.

Comment (Yisaer) [reply statistics/table.go]: updated.

**Review by time-and-fate — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**